src/main/java/io/antmedia/datastore/db/DataStore.java

```diff
@@ -424,6 +426,29 @@ public List<Token> listAllTokens (Map<String, String> tokenMap, String streamId,
 
 	public abstract boolean deleteToken (String tokenId);
 
+	/**
+	 * Whitelist specific token.
+	 * @param tokenId id of the token
+	 */
+	public abstract boolean whiteListToken(String tokenId);
+
+	/**
+	 * Get all blacklisted tokens.
+	 */
+	public abstract List<String> getBlackListedTokens();
+
+	/**
+	 * Delete all blacklisted expired tokens.
+	 */
+	public abstract Result deleteAllBlacklistedExpiredTokens(ITokenService tokenService);
+
+	/**
+	 * Whitelist all blacklisted tokens.
+	 *
+	 * @return
+	 */
+	public abstract boolean whiteListAllTokens();
```

Comment:
The concept of allowing only people with valid tokens is implicitly a whitelist, so I think you should not give the impression that there is a separate whitelist and a blacklist. Is it not only clearing the blacklist..?